### ISSUE TITLE
Setting a PIXI.Graphics object to be interactive does not work

### ISSUE THREAD

**Issue by n1313** [pixijs/pixi.js#821]
I draw a rectangle on stage, mark it as interactive and attach a click handler to it. When I click on the rectangle, I expect my handler to be fired. Yet, nothing happens. 

```
var stage = new PIXI.Stage(0xFFFFFF);

var renderer = PIXI.autoDetectRenderer(620, 380);

document.body.appendChild(renderer.view);

var graphics = new PIXI.Graphics();

graphics.beginFill(0xFF3300);
graphics.lineStyle(10, 0xffd900, 1);

graphics.drawRect(20, 20, 100, 200);
graphics.endFill();

graphics.interactive = true;
graphics.buttonMode = true;
graphics.click = function (e) {
    console.log(this, e);
}

stage.addChild(graphics);

requestAnimFrame(animate);

function animate() {
    requestAnimFrame(animate);
    renderer.render(stage);
}
```

Is this a bug or am I doing something wrong?

**Comment by Pavelyev** [comment]
Is this right that when hitArea is PIXI.Polygon it doesn't work? With rectangle it's OK.

        var stage = new PIXI.Stage(0xFFFFFF);

        var renderer = PIXI.autoDetectRenderer();
        document.getElementById('pixi').appendChild(renderer.view);

        var graphics = new PIXI.Graphics();

        var poly = new PIXI.Polygon(
            new PIXI.Point(50,100),
            new PIXI.Point(100,100),
            new PIXI.Point(100, 200),
            new PIXI.Point(50, 100)
        );

        graphics.beginFill(0xFF3300);
        graphics.lineStyle(4, 0xffd900, 1);

        graphics.drawPolygon(poly);

        graphics.hitArea = poly;
        graphics.interactive = true;
        graphics.buttonMode = true;
        graphics.click = function (e) {
            console.log(e);
        };
        
        stage.addChild(graphics);

        requestAnimationFrame(animate);
        function animate() {
            renderer.render(stage);
            requestAnimationFrame(animate);
        }

**Comment by englercj** [comment]
Polygon hit is broken, fix is in #3248
